src/errors.rs

```diff
@@ -160,3 +160,21 @@ impl From<serde_json::Error> for Error {
 }
 /// Convenient wrapper around std::Result.
 pub type Result<T> = ::std::result::Result<T, Error>;
+
+#[cfg(test)]
+mod tests {
+    #[test]
+    fn test_error_is_send_and_sync() {
```

Comment:
Typically this is done with `fn test_send_sync<T: Send + Sync>() test_send_sync::<Error>()`. Note that `fn very_useless_function` isn't actually needed here; the `impl SendSync for Error` will fail if the supertraits aren't supplied.

There's no way to make it fail at runtime rather than compile time though, unless doing a stupidly complex "faux specialization" trick which really isn't worth it; SendSync is a test you want to stop you immediately if you break it anyway.

Reply:
*Improved*